### PR TITLE
New readme.md + Add canRun function possibility for users

### PR DESCRIPTION
**Pull request by lepouletsuisse**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+##How to install Fabric3##
```
Comment (CSester) [inline readme.md]: Please use proper markdown Syntax

File: readme.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+1. First, be sure that Python 3.4+ is installed on your computer.
+2. Run "sudo python3 -m pip install Fabric3"
+3. Once it's done, be sure it's well installed by simply running "fab -V"
+
+##How to run Fabric3##
+
+1. Go in your project folder
+2. Clone the file https://github.com/CSester/automatic-deployment/blob/master/src/main.py in a src/ folder
```
Comment (CSester) [inline readme.md]: Why? Simply clone the repo, the script will be in the src subfolder...

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+3. Clone the repo http://idz-admin-app:7990/projects/DPLY/repos/deployment-configurations
```
Comment (CSester) [inline readme.md]: Remove this internal code. GITHUB IS PUBLIC !

File: src/main.py [hunk 4]
```diff
@@ -47,6 +47,30 @@ def main(configPath):
...
+          sys.path.append(local('pwd', capture=True))
+          import deploy
+          if 'canRun' not in dir(deploy):
+            print("No function canRun for deploy script in {}!".format(repo[0]))
+          else:
+            print("Function canRun exist for deploy script in {}!".format(repo[0]))
+            if deploy.canRun():
+              print("Deploy can run!")
+            else:
+              raise EnvironmentError("Can not continue, missing requirements for deploy script in {}! Aborting...".format(repo[0]))
```
Comment (CSester) [inline src/main.py:63]: Local repositories are not cleaned up when some canRun function fails !

File: src/main.py [hunk 5]
```diff
@@ -47,6 +47,30 @@ def main(configPath):
...
+          sys.path.remove(local('pwd', capture=True))
+          env['source-repository'] = repo[0]
```
Comment (CSester) [inline src/main.py]: Useless here... env is updated before runDeploy to keep the context

**Review by CSester — CHANGES_REQUESTED** [review]

**Comment by lepouletsuisse** [comment]
New commit done. 
Fix readme.md + add cleaning repo if canRun fail

File: readme.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+## How to install Fabric3 on Red Hat distribution ##
+
+1. First, be sure that Python 3.4+ is installed on your computer.
+2. Run "sudo python3 -m pip install Fabric3"
```
Comment (CSester) [inline readme.md]: Github markdown flavour also includes "code" blocks. See [this](https://guides.github.com/features/mastering-markdown/)

File: readme.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+3. Once it's done, be sure it's well installed by simply running "fab -V"
+
+## How to run Fabric3 ##
+
+1. Clone the repo https://github.com/CSester/automatic-deployment
+2. Create a new folder examples/ in the root directory
```
Comment (CSester) [inline readme.md]: The examples folder including the config.json file should already be in the repository. Please add necessary files

User should only edit tmpFolder (or some default value may be used ??)

File: readme.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+3. Create a new exemples/config.json file with following content:
```
Comment (CSester) [inline readme.md]: Typo: exEmples should be exAmples

File: readme.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    {
+    "source-commit": "test-branch",
+    "source-repository": "git@github.com:CSester/dummy-repo1.git",
+    "tmpFolder": ""
+    }
+4. Run "fab -f src/main.py main:exemples/config.json"
```
Comment (CSester) [inline readme.md]: Same typo as line 11

File: src/main.py [hunk 10]
```diff
@@ -47,6 +47,31 @@ def main(configPath):
...
+              print("Deploy can run!")
+            else:
+              raise EnvironmentError("Can not continue, missing requirements for deploy script in {}! Aborting...".format(repo[0]))
+          sys.path.remove(local('pwd', capture=True))
+      except ImportError as e:
+        print("No module deploy for {}!".format(repo[0]))
+        pass
+      except EnvironmentError as e:
+        local('rm -rf ../{}'.format(env.tmpFolder))
+        raise e
```
Comment (CSester) [inline src/main.py]: `raise` rather than `raise e`. This way the original stack trace is displayed

**Review by CSester — CHANGES_REQUESTED** [review]

File: readme.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+## How to install Fabric3 on Red Hat distribution ##
+
+1. First, be sure that Python 3.4+ is installed on your computer.
+2. Run ```sudo python3 -m pip install Fabric3```
+3. Once it's done, be sure it's well installed by simply running ```fab -V```
+
+## How to run Fabric3 ##
+
+1. Clone the repo for automatic deployment ```git clone https://github.com/CSester/automatic-deployment```
+2. Run ```fab -f src/main.py main:examples/config.json```
```
Comment (CSester) [inline readme.md]: The user should edit hosts, user and password fields.

The ```fab ... ``` command won't work as the user hasn't navigated to the automatic-deployment folder

**Review by CSester — CHANGES_REQUESTED** [review]

**Pull request merged**